internal/tutils/tutils.go

```diff
@@ -55,9 +55,14 @@ func GetResponseBody(t *testing.T, url string, auth *string) (int, string) {
 }
 
 func PostResponseBody(t *testing.T, url string, compose interface{}) (int, string) {
-	buf, err := json.Marshal(compose)
-	require.NoError(t, err)
-
+	var buf []byte
+	var err error
+	if compose == nil {
+		buf = []byte("{}") // Send an empty JSON object if compose is nil
+	} else {
+		buf, err = json.Marshal(compose)
+		require.NoError(t, err)
+	}
```

Comment:
Please do not send any body on `nil` argument. That is what I meant by zero-length body. This might be little bit tricky, let me explain. [NewRequest](https://pkg.go.dev/net/http#NewRequest) expects `io.Reader` so there are several options how you can achieve this:

* You can use an empty string: `strings.NewReader("")`
* Similarly byte slice that contains nothing: `bytes.NewReader([]byte)`
* Or perhaps `bytes.NewReader(nil)` does the same thing
* To my surprise, there is no `io.NullReader` in the standard library, there is `ioutil.Discard` do, a writer that does nothing.

Reply:
what is the different between sending strings.NewReader("") or ComposeBlueprintJSONBody{} both of them bring this result

Reply:
The former sends nothing (no body). The latter sends '{}' in the body.

You can also define a helper type to do the same:

```go
type eofReader struct{}

func (eofReader) Read([]byte) (int, error) {
	return 0, io.EOF
}

var EOFReader io.Reader = eofReader{}
```

When you pass `EOFReader` it immediately returns EOF (end of file), so no body is sent. Does it make sense?